src/ExperimentationService.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as fse from "fs-extra";
+import * as vscode from "vscode";
+import { addContextProperty, sendInfo } from "vscode-extension-telemetry-wrapper";
+import { getExperimentationService, IExperimentationService, IExperimentationTelemetry, TargetPopulation } from "vscode-tas-client";
+
+class ExperimentationTelemetry implements IExperimentationTelemetry {
+
+    public setSharedProperty(name: string, value: string): void {
+        addContextProperty(name, value);
+    }
+
+    public postEvent(eventName: string, props: Map<string, string>): void {
+        const data: any = {};
+        data.__event_name__ = eventName;
+        for (const [property, value] of props) {
+            data[property] = value;
+        }
+        sendInfo("", data);
+    }
+}
+
+let expService: IExperimentationService;
+
+export function getExpService() {
+    return expService;
+}
+
+export async function init(context: vscode.ExtensionContext): Promise<void> {
+    const packageJson: {} = await fse.readJSON(context.asAbsolutePath("package.json"));
+    // tslint:disable: no-string-literal
+    const extensionName = `${packageJson["publisher"]}.${packageJson["name"]}`;
+    const extensionVersion = packageJson["version"];
+    // tslint:enable: no-string-literal
+    expService = getExperimentationService(extensionName, extensionVersion,
+        TargetPopulation.Public, new ExperimentationTelemetry(), context.globalState);
+
+    // Due to a bug in the tas-client module, a call to isFlightEnabledAsync is required to begin
```

Comment:
You should be able to drop these dummy checks

Reply:
I tried to remove these lines. But I found that when I'm calling `getExpService().isCachedFlightEnabled("defaultMaven");`, the cached features are empty.

![image](https://user-images.githubusercontent.com/6193897/81761709-3977a300-94fd-11ea-8b5e-225719526224.png)

Within the dummy code, the cached features are correct.

_// The vscode-tas-client I'm using is 0.0.659_

Reply:
Oh interesting. Maybe we still have another bug to track down. OK, keep these checks for now and I'll look into the issue ASAP, and with the next update to the package, you can remove them.